core/story_service.go

```diff
@@ -79,7 +92,36 @@ func (s *StoryService) formatStoryInPRTitle(story string) string {
 	return fmt.Sprintf("[%s]", story)
 }
 
-func (s *StoryService) enrichBody(rawBody, _ string) string {
-	// TODO(claire.philippe): to implement
-	return rawBody
+func (s *StoryService) enrichBody(rawBody, story string) (string, error) {
+	if story == "" || !BodyEnrichmentEnabled() {
+		return rawBody, nil
+	}
+
+	link, err := s.formatBodyInPRTitle(story)
+	if err != nil {
+		return "", fmt.Errorf("could not enrich the body with the Story: %w", err)
+	}
+
+	if rawBody == "" {
+		return link, nil
+	}
+
+	return strings.Join([]string{rawBody, link}, "\n\n"), nil
+}
+
+func (s *StoryService) formatBodyInPRTitle(story string) (string, error) {
+	tool := GetStoryTool()
+	urlTemplate, ok := urlPatterns[tool]
+	if !ok {
+		availableTools := []string{}
+		for tool := range urlPatterns {
+			availableTools = append(availableTools, tool)
+		}
+		return "", fmt.Errorf("tool set in config (%s) doesn't match possible values (%s)", tool, availableTools)
+	}
+
+	baseUrl := GetStoryToolBaseUrl()
+	url := fmt.Sprintf(urlTemplate, baseUrl, story)
+
+	return fmt.Sprintf("[%s](%s)", storyLinkValue, url), nil
```

Comment:
could we put the ticket number in there? and put the tool name instead of "Story"? On my side for instance I would be thrilled if I see "Linear [WARP-1257]". And the last thing I would love (but it's a nit) is the ability to actually put that in a bullet point list (it could be optional). My PR descriptions always look like this:

```
  - Linear link
  - Implement blalbabla
  ...
```